scripts/build/.variables

```diff
@@ -86,7 +86,7 @@ if [ "$CGO_ENABLED" = "1" ] && [ "$GO_LINKMODE" = "static" ]; then
     GO_BUILDTAGS="$GO_BUILDTAGS osusergo"
 fi
 if [ -n "$GO_STRIP" ]; then
-    GO_LDFLAGS="$GO_LDFLAGS -s"
+    GO_LDFLAGS="$GO_LDFLAGS -s -w"
```

Comment:
As we now only set `-w` when `GO_STRIP` (which looks to be set through `STRIP_TARGET` in the bakefile) is enabled, do we still need the `llvm-strip` hack?

If "not stripping" is the fix for Darwin, could we instead 

- document this portion (and the corresponding `STRIP_TARGET`) to describe that it's not supported for Darwin binaries? (when using llvm < 12)
- or unset `-w` if the target is darwin (or make sure `STRIP_TARGET` won't be set

Reply:
> do we still need the `llvm-strip` hack?

yes this is a bug with llvm that has been fixed in llvm 12 and only occurs with `darwin/amd64`. not sure why `llvm-strip` is always called even if `-s` and `-w` flags are not specified at all. I also can't repro that behavior with alpine base image because it uses llvm 12.

> If "not stripping" is the fix for Darwin, could we instead
> 
>     * document this portion (and the corresponding `STRIP_TARGET`) to describe that it's not supported for Darwin binaries? (when using llvm < 12)
> 
>     * or unset `-w` if the target is darwin (or make sure `STRIP_TARGET` won't be set

yes a comment lgtm, will do that.

Reply:
> not sure why llvm-strip is always called even if -s and -w flags are not specified at all.

Ah, right; so removing `-s` and/or `-w` for `darwin/amd64` won't work around the issue; gotcha.